hpct-ldap-server/src/managers/openldap.py

```diff
@@ -0,0 +1,491 @@
+            rc = subprocess.call(
+                ["update-ca-certificates"], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
+            )
+
+            if rc != 0:
+                raise Exception("Unable to add CA certificate to trusted certs")
+
+            # Get Server Hostname
+            hostname = subprocess.run(["cat", "/etc/hostname"], capture_output=True, text=True).stdout.strip()
+
+            # Private Key for Server
+            pks_args = [
+                "certtool",
+                "--generate-privkey",
+                "--bits",
+                "2048",
+                "--outfile",
+                f"/etc/ldap/ldap-{hostname}_slapd_key.pem",
+            ]
+            rc = subprocess.call(pks_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to create private key for server.")
+
+            # Template for Server Certificate
+            sc_template = [
+                f"organization = {org}",
+                f"cn = {hostname}",
+                "tls_www_server",
+                "encryption_key",
+                "signing_key",
+                "expiration_days = 365",
+            ]
+            sc_template = "\n".join(sc_template)
+            with open(f"/etc/ssl/ldap-{hostname}.info", "w") as f:
+                f.write(f"{sc_template}")
+
+            # Create Server Certificate
+            sc_args = [
+                "certtool",
+                "--generate-certificate",
+                "--load-privkey",
+                f"/etc/ldap/ldap-{hostname}_slapd_key.pem",
+                "--load-ca-certificate",
+                "/etc/ssl/certs/mycacert.pem",
+                "--load-ca-privkey",
+                "/etc/ssl/private/mycakey.pem",
+                "--template",
+                f"/etc/ssl/ldap-{hostname}.info",
+                "--outfile",
+                f"/etc/ldap/ldap-{hostname}_slapd_cert.pem",
+            ]
+            rc = subprocess.call(sc_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to create server certificate.")
+
+            # Adjust Permissions and Ownership
+            shutil.chown(f"/etc/ldap/ldap-{hostname}_slapd_key.pem", group="openldap")
+            os.chmod(f"/etc/ldap/ldap-{hostname}_slapd_key.pem", 0o640)
+
+            # Server Certificates LDIF
+            sc_ldif = [
+                "dn: cn=config",
+                "add: olcTLSCACertificateFile",
+                "olcTLSCACertificateFile: /etc/ssl/certs/mycacert.pem",
+                "-",
+                "add: olcTLSCertificateFile",
+                f"olcTLSCertificateFile: /etc/ldap/ldap-{hostname}_slapd_cert.pem",
+                "-",
+                "add: olcTLSCertificateKeyFile",
+                f"olcTLSCertificateKeyFile: /etc/ldap/ldap-{hostname}_slapd_key.pem",
+            ]
+            sc_ldif = "\n".join(sc_ldif)
```

Comment:
Could use "(" and ")" with newlines instead of list and join.

Reply:
Updated as suggested